### ISSUE TITLE
Fallback when the key is not in the cache

### ISSUE THREAD

**Issue by christophemenager** [phrase/react_native_sdk_example#4]
Sorry, it's me again ! 

I successfully integrated the react-native SDK into my project, thanks for the hard work @egze @theSoenke @DarkMinstrel  :)

I have a last question : If the translation key is not found in the translations cache of Phrase, is it possible to fallback to the translation key embedded in the app ? Currently it works for the whole translation file, but not on a key granularity.

This implies that once I start creating my first OTA release (for example to fix some translation wordings), I am forced to keep creating releases each time I make a change. And users will still probably see missing translations as the refresh is made in the background, and cache is used as default.

In a nutshell : once you have made a release for all users, how do you keep adding new keys ?

**Comment by christophemenager** [comment]
Just to let you know, I have explained the issue for the support team, maybe a little bit clearer : 

I have a functional issue with the integration of OTA on react-native, using your recent sdk 1.0.0 (I followed the steps  on the example repo)

Two issues with this integration : 
1. Fetch of new OTA releases is made after startup of the app, which means you have to start the app a second time to see new translations
2. OTA cache (of present) is the only source of truth, and is not merged with local translation file. This means you are forced to continue making releases every-time there are new keys, for the lifetime of the product.

Maybe I missed something but I don't get how to use OTA on react-native properly. For the moment, it looks like an extra constraint more than an actual benefit...

**Comment by christophemenager** [comment]
As it may help others, I solved this issue in two steps : 
1. Did not use backends from i18next (chainedBackend do not support merge of two backends), I simply call `phrase.requesTranslation` and then do `i18next.addResourceBundle`
2. I patched the react-native-phrase-sdk to await the `updateFromApi`, in order not to have to relaunch the application to actually get the update. This fits my need perfectly, maybe not the case for all users

**Comment by egze** [comment]
Hi @christophemenager 

Glad you could solve your issue, and thanks for posting your findings.

I just want to comment on point 2 about waiting for the API response before starting the app. With slow or unstable internet, it will hurt the user experience, so the approach should be used with care.

**Comment by christophemenager** [comment]
Thanks for your warning @egze , you are perfectly right it's maybe not the best UX ever :) 

I changed it a bit : 
1. During startup I check in the cache
2. After startup, I check on the API and re-render screens if needed thanks to the option ```
react: {
    bindI18nStore: "added",
  },
``` added in i18next options

This way we can have a good balance between fast startup in poor internet conditions and up-to-date translations (with a slight delay in the UX, not an issue in my case)

Could be nice to expose two methods in the API of your SDK, one for reading the cache, one for checking the API. This offers more flexibility of integration for different use cases

**Comment by RRaideRR** [comment]
Sorry for being late to the party, but I am right now in a similar situation and wondering how do you guys develop new features?

Example: I have a new feature in my app that uses language key "intro.helloWorld" with translation "Hello World.". I build a new version of the app, with the updated local language files, and deploy it to the stores.

A user is updating the app. [According to this graph](https://help.phrase.com/help/phrase-over-the-air-introduction), the following procedure is then done on his app start:
1. Is this the first launch of the app? It isn't because the app was (with a older version) launched before.
2. So we are taking the translations from the latest available translation cache.
3. The user just sees the key "intro.helloWorld" in the app but not the translation.
4. In the background we fetch the latest translations which will be shown to the user on the next app start.

Am I making something wrong? I cannot see how I can develop new features and ship them flawlessly with this process.
